### PR TITLE
[#IOPID-127] Porting `content-digest` crypto fn from `io-functions-lollipop`

### PR DESCRIPTION
**Pull request by BurnedMarshal**
#### List of Changes

Copy crypto functions from `io-functions-lollipop` and `io-backend`.

#### Motivation and Context

Avoid code duplication across repos.

#### How Has This Been Tested?

unit test

#### Types of changes

- [X] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/utils/crypto.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+export const validateDigestHeader = (
+  contentDigestHeader: string,
+  body: Buffer | string
+): void => {
+  if (!contentDigestHeader) {
+    throw new Error("Content-Digest header missing");
+  }
+
+  // Validate
+  const contentDigestPattern = new RegExp("(.+)=:(.+):");
```
Comment (BurnedMarshal) [inline src/utils/crypto.ts:74]: We can reduce the regex vulnerability with this
```suggestion
  const contentDigestPattern = new RegExp("(sha-256|sha-512)=:(.+):");
```

File: src/utils/crypto.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import * as crypto from "crypto";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+
+/**
+ * Returns hashed input based on sha256 algo
+ */
+export const toHash = (s: string): NonEmptyString =>
```
Comment (gquadrati) [inline src/utils/crypto.ts:7]: Maybe `toSha256` is more accurate? @michaeldisaro

**Review by gquadrati — APPROVED** [review]
Just a small comment. Agree with regex refactor.

File: src/utils/crypto.ts (hunk 2 above)
Comment (BurnedMarshal) [reply src/utils/crypto.ts]: We changed the fn name to `sha256` to keep the same name used into the `io-backend`.

**Pull request merged**